src/shogun/mathematics/linalg/LinalgNamespace.h

```diff
@@ -114,20 +109,26 @@ LinalgBackendBase* infer_backend(const Container<T>& a, const Container<T>& b)
 template <typename T>
 void to_gpu(SGVector<T>& a, SGVector<T>& b)
 {
-	m_gpu_transfer.lock();
-
-	REQUIRE(!a.on_gpu(), "The vector is already on GPU.\n");
-	LinalgBackendBase* gpu_backend = sg_linalg->get_gpu_backend();
+	sg_linalg->m_gpu_transfer.lock();
 
-	if (gpu_backend)
-		b = SGVector<T>(gpu_backend->to_gpu(a), a.vlen);
+	if (a.on_gpu())
+	{
+		SG_SWARNING("The vector is already on GPU.\n");
+	}
 	else
 	{
-		SG_SWARNING("Trying to access GPU memory without GPU backend registered.\n");
-		b = a;
+		LinalgBackendBase* gpu_backend = sg_linalg->get_gpu_backend();
+
+		if (gpu_backend)
+			b = SGVector<T>(gpu_backend->to_gpu(a), a.vlen);
+		else
+		{
+			SG_SWARNING("Trying to access GPU memory without GPU backend registered.\n");
```

Comment:
I know this was there before but I just reaslied this will cause tons of warnings for gpu code being executed without a gpu present. I wonder whether that is good?